msal/src/main/java/com/microsoft/identity/client/ISingleAccountPublicClientApplication.java

```diff
@@ -81,9 +84,65 @@
      * @link AuthenticationCallback#onError(MsalException)}.
      */
     void signIn(@NonNull final Activity activity,
-                @NonNull final String loginHint,
+                @Nullable final String loginHint,
                 @NonNull final String[] scopes,
-                @NonNull final AuthenticationCallback callback);
+                @NonNull final AuthenticationCallback callback
+    );
+
+    /**
+     * Allows a user to sign in to your application with one of their accounts. This method may only
+     * be called once: once a user is signed in, they must first be signed out before another user
+     * may sign in. If you wish to prompt the existing user for credentials use
+     * {@link #reauthorize(Activity, String[], Prompt, AuthenticationCallback)} or
+     * {@link #acquireToken(AcquireTokenParameters)}.
+     * <p>
+     * Note: The authority used to make the sign in request will be either the MSAL default: https://login.microsoftonline.com/common
+     * or the default authority specified by you in your configuration
+     *
+     * @param activity  Non-null {@link Activity} that is used as the parent activity for launching the {@link com.microsoft.identity.common.internal.providers.oauth2.AuthorizationActivity}.
+     * @param loginHint Optional. If provided, will be used as the query parameter sent for authenticating the user,
+     *                  which will have the UPN pre-populated.
+     * @param scopes    The non-null array of scopes to be consented to during sign in.
+     *                  MSAL always sends the scopes 'openid profile offline_access'.  Do not include any of these scopes in the scope parameter.
+     *                  The access token returned is for MS Graph and will allow you to query for additional information about the signed in account.
+     * @param callback  {@link AuthenticationCallback} that is used to send the result back. The success result will be
+     *                  sent back via {@link AuthenticationCallback#onSuccess(IAuthenticationResult)}.
+     *                  Failure case will be sent back via {
+     * @link AuthenticationCallback#onError(MsalException)}.
+     */
+    void signIn(@NonNull final Activity activity,
```

Comment:
@shoatman I have introduced this overload for apps who wish to use SA-PCA, but _also_ wish to specify a `prompt` for the "first run"

Reply:
I think that's fine... if we do this... if the second method actually required still?

Reply:
@shoatman Technically, the second method was _never_ required since `acquireToken(params)` can do all the same things.... That said....

I see this overload of `signIn` as having a different purpose from the _original `signIn()` and the newly added `reauthorize()` (soon to be renamed, `signInAgain()`
- `signIn` _without_ a prompt param will allow for SSO state pickup from the browser, if the user already has state
- `signIn` _with_ a prompt param allows the developer to tweak the prompt behavior for the _first run_, but just like the other `signIn()` method, also disalllows multiple invocations of `signIn()`
- `signInAgain()` accepts no account _at all_, and can only be used once a user has already signed in -- it's purpose is only if you must reprompt the user for creds, as a banking app might upon relaunch